pkg/cloud/cloud.go

```diff
@@ -1024,6 +1024,15 @@ func (c *cloud) WaitForAttachmentState(ctx context.Context, volumeID, expectedSt
 		// but DescribeVolume told us volume is detached, we will short-circuit this long wait loop and return error
 		// so as AttachDisk can be retried without waiting for 20 minutes.
 		if (expectedState == volumeAttachedState) && alreadyAssigned && (attachmentState != expectedState) {
+			request := &ec2.AttachVolumeInput{
+				Device:     aws.String(expectedDevice),
+				InstanceId: aws.String(expectedInstance),
+				VolumeId:   aws.String(volumeID),
+			}
+			_, err := c.ec2.AttachVolume(ctx, request)
+			if err != nil {
+				klog.InfoS("WaitForAttachmentState: AttachVolume call error", "volumeID", volumeID, "instanceID", expectedInstance, "Device", expectedDevice, "err", err)
```

Comment:
add a return to this case so the error is passed upstream to the user (this can help them if the error is their fault, such as trying to attach a bx volume to a non-bx instance)

Reply:
Added return